src/fragments/lib/auth/js/advanced.mdx

```diff
@@ -169,72 +169,59 @@ const App = () => {
 
 ### Google sign-in (React)
 
+
 ```js
 import React, { useEffect } from 'react';
+import jwt from 'jwt-decode'
 import { Auth } from 'aws-amplify';
-// To federated sign in from Google
-const SignInWithGoogle = () => {
 
+const SignInWithGoogle = () => {
     useEffect(() => {
-        const ga = window.gapi && window.gapi.auth2 ? 
-            window.gapi.auth2.getAuthInstance() : 
-            null;
-
-        if (!ga) createScript();
-    }, [])
-
-    const signIn = () => {
-        const ga = window.gapi.auth2.getAuthInstance();
-        ga.signIn().then(
-            googleUser => {
-                getAWSCredentials(googleUser);
-            },
-            error => {
-                console.log(error);
-            }
-        );
-    }
-
-    const getAWSCredentials = async (googleUser) => {
-        const { id_token, expires_at } = googleUser.getAuthResponse();
-        const profile = googleUser.getBasicProfile();
-        let user = {
-            email: profile.getEmail(),
-            name: profile.getName()
-        };
-        
-        const credentials = await Auth.federatedSignIn(
-            'google',
-            { token: id_token, expires_at },
-            user
-        );
-        console.log('credentials', credentials);
-    }
+        /// Check for previous Google client initialization
+        if (!window.google && !window.google?.accounts) createScript();
+    }, []);
 
+    // Load the Google client
     const createScript = () => {
-        // load the Google SDK
         const script = document.createElement('script');
-        script.src = 'https://apis.google.com/js/platform.js';
+        script.src = 'https://accounts.google.com/gsi/client';
         script.async = true;
-        script.onload = initGapi;
+        script.defer = true;
+        script.onload = initGsi;
         document.body.appendChild(script);
     }
 
-    const initGapi = () => {
-        // init the Google SDK client
-        const g = window.gapi;
-        g.load('auth2', function() {
-            g.auth2.init({
-                client_id: 'your_google_client_id',
-                // authorized scopes
-                scope: 'profile email openid'
-            });
+    /// Initialize Google client and render Google button
+    const initGsi = () => {
+        window.google?.accounts.id.initialize({
```

Comment:
Since it's used twice maybe we can just wrap both calls inside a
```
if (window.google) { }
```
maybe?

Reply:
Also maybe include `accounts` in the check for consistency